remotes/docker/config/hosts.go

```diff
@@ -232,11 +232,15 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 				rhosts[i].Authorizer = authorizer
 			}
 
-			// When TLS has been explicitly configured for the operation or host, use the
-			// docker.HTTPFallback roundtripper to catch TLS errors and re-attempt the request as http.
-			// This allows preference for https when configured but also catches TLS errors early enough
-			// in the request to avoid sending the request twice or consuming the request body.
-			if host.scheme == "http" && explicitTLS {
+			// When TLS has been explicitly configured for the operation or host and
+			// the protocol from the port number is ambiguous, use the
+			// docker.HTTPFallback roundtripper to catch TLS errors and re-attempt the
+			// request as http. This allows preference for https when configured but
+			// also catches TLS errors early enough in the request to avoid sending
+			// the request twice or consuming the request body.
+			if host.scheme == "http" && explicitTLS &&
+				strings.ContainsAny(host.host, ":") &&
```

Comment:
This matches IPv6 addresses too

Reply:
I'll add relevant test cases and will probably need to use the actual parsing to decipher the port